src/services/sources.ts

```diff
@@ -28,7 +27,7 @@ export const sourcesApi = createApi({
         body,
       }),
     }),
-    createSource: builder.mutation<MetadataResponse<{name: string; namespace?: string}>, SourceWithRepository>({
+    createSource: builder.mutation<CreateSourceResult, SourceWithRepository>({
```

Comment:
I would prefer to use it like this:

1. Change `MetadataResponse` to:
```
export type MetadataResponse<Meta = {}, Spec = {}> = {
  metadata: Meta;
  spec?: Spec;
};
```

2. Use it as:
```
MetadataResponse<
  {
    name: string;
    namespace: string;
    uid: string;
    resourceVersion: string;
    generation: 1;
    creationTimestamp: string;
  },
  {repository: Repository}
>
```

Reply:
`MetadataResponse` should have only metadata, not the `spec`. It could be optionally used though like:
```
MetadataResponse<{
    name: string;
    namespace: string;
    uid: string;
    resourceVersion: string;
    generation: 1;
    creationTimestamp: string;
  }> & {spec: { repository: Repository }}
```

But I don't think that it makes sense, given that it's just wrapping in `meta`

Reply:
Actually, I see that the `spec` is already used anyway in `MetadataResponse`. Will use it then, although I think that `MetadataResponse` should be rethinked later, so it will be helpful 👍

Reply:
Let's leave it for now then. I will create smth like `MetaSpecResponse` with 2 arguments for generic `Meta` and `Spec`. But I don't think it's a big problem to have `meta` and `spec` fields typed in `MetadataResponse` as those 2 fields are always come together.

Reply:
@Toxer10, basically, this `MetadataResponse` is `Definition` :) Because it's CRD's object definition. It doesn't have to have `metadata`, it might be custom, and same for spec.